File: 2021/09/26/next_greatest_letter.py
"""
https://leetcode.com/problems/find-smallest-letter-greater-than-target/
Author: Andrew Jarombek
Date: 9/26/2021
"""

# Back home.  Without wanting you to feel bad or anything, I just want you to know that I miss you a lot.
# My love is always here for you.


from typing import List


def next_greatest_letter(letters: List[str], target: str) -> str:
    for letter in letters:
        if letter > target:
            return letter

    return letters[0]


if __name__ == '__main__':
    assert next_greatest_letter(["a", "c", "f"], "b") == "c"
    assert next_greatest_letter(["a", "c", "f"], "f") == "a"

File: 2022/12/26/nextGreatestLetter.js
/**
 * https://leetcode.com/problems/find-smallest-letter-greater-than-target/
 * @author Andrew Jarombek
 * @since 12/26/2022
 */

const nextGreatestLetter = function(letters, target) {
    for (let letter of letters) {
        if (letter > target) {
            return letter;
        }
    }

    return letters[0];
};

console.assert(nextGreatestLetter(["c","f","j"], "a") === "c");
console.assert(nextGreatestLetter(["x","x","y","y"], "z") === "x");
